api/classInstances/classInstanceMiddleware.js

```diff
@@ -25,107 +20,34 @@ const checkClassInstanceExist = async (req, res, next) => {
-  if (!start_date) next({ status: 400, message: 'start_date is required' });
-  if (!end_date) next({ status: 400, message: 'end_date is required' });
-  if (!location) next({ status: 400, message: 'location is required' });
-  if (typeof size !== 'number')
-    next({ status: 400, message: 'size must be of type number' });
-  if (typeof open_seats_remaining !== 'number')
-    next({
-      status: 400,
-      message: 'open_seats_remaining must be of type number',
-    });
-  if (typeof instructor_id !== 'number')
-    next({ status: 400, message: 'instructor_id must be of type number' });
-  if (typeof course_type_id !== 'number')
-    next({ status: 400, message: 'course_type_id must be of type number' });
-  if (typeof start_time !== 'string')
-    next({ status: 400, message: 'start_time must be of type string' });
-  if (typeof end_time !== 'string')
-    next({ status: 400, message: 'end_time must be of type string' });
-  if (typeof start_date !== 'string')
-    next({ status: 400, message: 'start_date must be of type string' });
-  if (typeof end_date !== 'string')
-    next({ status: 400, message: 'end_date must be of type string' });
-  if (typeof location !== 'string')
-    next({ status: 400, message: 'location must be of type string' });
-  if (size < 0 || size >= 2 ** 31)
-    next({
-      status: 400,
-      message: 'size must be at least zero and less than 2^31',
-    });
-  if (open_seats_remaining <= -(2 ** 31) || open_seats_remaining >= 2 ** 31)
-    next({
-      status: 400,
-      message:
-        'open_seats_remaining must not have an absolute value that exceeds 2^31',
-    });
-  if (verifyTimeWithoutTimeZone(start_time))
-    next({
-      status: 400,
-      message: 'start_time must be in the HH:MM:SS format and be a valid time',
-    });
-  if (verifyTimeWithoutTimeZone(end_time))
-    next({
-      status: 400,
-      message: 'end_time must be in the HH:MM:SS format and be a valid time',
-    });
-  if (verifyDate(start_date))
-    next({
-      status: 400,
-      message:
-        'start_date must be in the YYYY-MM-DD format and be a valid date',
-    });
-  if (verifyDate(end_date))
-    next({
-      status: 400,
-      message: 'end_date must be in the YYYY-MM-DD format and be a valid date',
-    });
-  if (location.length > 255)
-    next({
-      status: 400,
-      message: 'the location string must not exceed a length of 255 characters',
-    });
-
-  const instructor = await findByInstructorId(instructor_id);
-  if (!instructor)
-    next({
-      status: 400,
-      message: `instructor with id ${instructor_id} does not exist`,
-    });
-  const courseType = await getAllCourseTypes()
-    .where({ course_type_id })
-    .first();
-  if (!courseType)
-    next({
-      status: 400,
-      message: `courseType with id ${course_type_id} does not exist`,
+const validateClassObject = async (req, res, next) => {
+  try {
+    const validated = await classSchema.validate(req.body, {
```

Comment:
Great job utilizing `yup` to validate classes. It reduces a lot of code, accomplishes the same goal, is way more readable, and formats a class to be the correct way. Only thing I would change is the name "validated" for the validated class. I would name it "validatedClass" just to make it a bit more clear.

Reply:
Great point! I will update it :)